### PR TITLE
WPML Integration - fix more rewrite URLs translation

### PR DESCRIPTION
**Pull request by lucatume**
[TEC-4689](https://theeventscalendar.atlassian.net/browse/TEC-4689)
[TEC-4360](https://theeventscalendar.atlassian.net/browse/TEC-4360) 

Based on [Common PR](https://github.com/the-events-calendar/tribe-common/pull/1897)

Artifacts:
* tests where possible; I've not gone and fully mocked WPML or pull the plugin
* [screencast for TEC-4360](https://share.cleanshot.com/0P7gp47V)
* [screencast for TEC-4689](https://share.cleanshot.com/zxvhKkJJ)

[TEC-4689]: https://theeventscalendar.atlassian.net/browse/TEC-4689?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TEC-4360]: https://theeventscalendar.atlassian.net/browse/TEC-4360?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Integrations/WPML/Rewrites.php [hunk 1]
```diff
@@ -207,9 +233,188 @@ public function filter_tax_base_slug( $bases, $method ) {
...
 	}
+
+	/**
+	 * Translate the Event single slugs.
+	 *
+	 * @param array<string,array<string>> $bases The bases to translate.
+	 *
+	 * @return array<string,array<string>> The translated bases.
+	 */
+	protected function translate_single_slugs( array $bases ) {
```
Comment (stratease) [inline src/Tribe/Integrations/WPML/Rewrites.php:249]:
```suggestion
	protected function translate_single_slugs( array $bases ): array {
```

File: src/Tribe/Integrations/WPML/Rewrites.php [hunk 2]
```diff
@@ -207,9 +233,188 @@ public function filter_tax_base_slug( $bases, $method ) {
...
+	 * Translate the Event archive slugs.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,array<string>> $bases The bases to translate.
+	 *
+	 *
+	 * @return array<string,array<string>> The translated bases.
+	 */
+	protected function translate_archive_slugs( array $bases ) {
```
Comment (stratease) [inline src/Tribe/Integrations/WPML/Rewrites.php:292]:
```suggestion
	protected function translate_archive_slugs( array $bases ): array {
```

File: src/Tribe/Integrations/WPML/Rewrites.php [hunk 3]
```diff
@@ -207,9 +233,188 @@ public function filter_tax_base_slug( $bases, $method ) {
...
+	/**
+	 * Filters the bases used to generate TEC rewrite rules to use WPML managed translations.
+	 *
+	 * @param array<string,string>  $bases  An array of bases to translate.
+	 * @param string $method The method used to generate the rewrite rules, unused by this method.
+	 * @param array<string>  $domains
+	 *
+	 * @return array An array of bases each with its (optional) WPML managed translations set.
+	 */
+	public function filter_tribe_events_rewrite_i18n_slugs_raw( $bases, $method, $domains ) {
```
Comment (stratease) [inline src/Tribe/Integrations/WPML/Rewrites.php:330]:
```suggestion
	public function filter_tribe_events_rewrite_i18n_slugs_raw( $bases, $method, $domains ): array {
```

**Review by stratease — APPROVED** [review]
Couple minor typing recommendations if valid.

File: tests/wpunit/Tribe/Events/RewriteTest.php [hunk 4]
```diff
@@ -300,6 +282,25 @@ public function it_urls() {
 		];
 	}
 
+	/**
+	 * It should correctly handle translated rules
+	 *
+	 * @test
+	 * @dataProvider it_urls_data_provider
+	 */
+	public function should_correctly_handle_translated_rules( $path, $expected_path ) {
```
[the remark this answers is not in the document]
Comment (Camwyn) [reply tests/wpunit/Tribe/Events/RewriteTest.php:291]: Might as well do it here too 😉

**Review by Camwyn — CHANGES_REQUESTED** [review]

**Pull request merged**